typings/typescript.d.ts

```diff
@@ -0,0 +1,8 @@
+import "typescript";
+
+declare module "typescript" {
+	export function isExpression(node: Node): node is ts.Expression;
+	export function isLeftHandSideExpression(
+		node: Node
+	): node is ts.LeftHandSideExpression;
+}
```

Comment:
We shouldn't need this anymore. If there's any API that still relies on TypeScript's privates, let's file it as a followup issue. I'm not sold that we should be relying on them in this way.

Reply:
`isExpression` is still used by `isConciseBody` and `isForInitializer`.

`LeftHandSideExpression` is still used by `isAssignmentExpression`.
The following use `isAssignmentExpression` and thus also use `LeftHandSideExpression`.

- `isEqualsAssignmentExpression`
- `isArrayDestructuringAssignment`
- `isObjectDestructuringAssignment`
- `isDestructuringAssignment`
- `isArrayBindingOrAssignmentElement`
- `isBindingOrAssignmentElement`

We'd need to drop all these type guards if we don't use these private methods.

Reply:
Yeah let's drop them for now. We can always add them later if the privates are worth it. But I'd rather not poke into TS internals.

Once this PR goes in we can file a new issue on the TS repo to expose more things since it's been so long since this was discussed.